### ISSUE TITLE
fatal: no submodule mapping found in .gitmodules for path 'lib/caffe-tpp-net-python'

### ISSUE THREAD

**Issue by csantosbh** [zhujiagang/DTPP#3]
Hello,

Thank you for publishing the code for your work!
I'm trying to reproduce it, but had an issue. When trying to clone the DTPP repository, I get the following error:

```
fatal: no submodule mapping found in .gitmodules for path 'lib/caffe-tpp-net-python'
```

Browsing the `lib` folder on github, I see that the caffe-tpp-net-python is not linked to any repository. Perhaps there was an error in the declaration of this submodule?

**Comment by zhujiagang** [comment]
Yes, I've also found the error and have updated README.md. You just have to `cp -r caffe-tpp-net/ caffe-tpp-net-python/`.

**Comment by csantosbh** [comment]
Nice, thank you! I can confirm that the issue has been solved. Great job!

I have one suggestion related to the clone issue. It appears that you can only clone if you have the correct SSH keys configured for github, due to the fact that the submodules are cloned using SSH.

That can be fixed by using the HTTP URLs for the submodule repos. These commands worked locally:

```
git config submodule.lib/caffe-tpp-net.url http://github.com/zhujiagang/caffe-tpp-net
git config submodule.lib/dense-flow.url http://github.com/yjxiong/dense_flow
```

However, they do not change any local file that can be committed. If you wish to perform this configuration in a way that you can persist them to the remote (that is, to github), [this thread](https://stackoverflow.com/questions/913701/how-to-change-the-remote-repository-for-a-git-submodule) seems to explain how this can be done.

Thanks again!
